tests/unit/test_cli.py

```diff
@@ -491,16 +495,49 @@ def test_mv_mix_secrets_folders(cli_runner, vault_with_token):
     assert result.exit_code != 0
 
 
-def test_template(cli_runner, vault_with_token):
+def test_template_from_stdin(cli_runner, vault_with_token):
     vault_with_token.db = {"a/b": {"value": "c"}}
 
     result = cli_runner.invoke(
         cli.cli, ["template", "-"], input="Hello {{ vault('a/b') }}"
     )
 
+    print(result, file=sys.stdout)
+    assert result.stdout == "Hello c"
+    assert result.exit_code == 0
+
+
+def test_template_from_file(cli_runner, vault_with_token):
+    vault_with_token.db = {"a/b": {"value": "c"}}
+
+    with tempfile.NamedTemporaryFile(mode="w+") as fp:
+        fp.write("Hello {{ vault('a/b') }}")
+        fp.flush()
+        result = cli_runner.invoke(
+            cli.cli, ["template", fp.name], catch_exceptions=False
+        )
+
     assert result.exit_code == 0
     assert result.stdout == "Hello c"
 
+def test_template_from_file_with_include(cli_runner, vault_with_token):
+    vault_with_token.db = {"a/b": {"value": "c"}}
+
+    tmp_dir = tempfile.mkdtemp(dir=os.getcwd())
+    try:
+        os.mkdir(os.path.join(tmp_dir, 'foobar'))
+        with tempfile.NamedTemporaryFile(dir=tmp_dir, mode="w+") as file1:
+            file1.write("Hello {{ vault('a/b') }}\n{% include('" + file1.name + "') %}")
```

Comment:
Are we including a file in itself ? Doesn't this cause some sort of infinite recursion ?